pkg/resourcemanager/controller/health/progressing/add.go

```diff
@@ -80,9 +81,11 @@ func (r *Reconciler) AddToManager(ctx context.Context, mgr manager.Manager, sour
 	}
 
 	for resource, obj := range map[string]client.Object{
-		"deployments":  &appsv1.Deployment{},
-		"statefulsets": &appsv1.StatefulSet{},
-		"daemonsets":   &appsv1.DaemonSet{},
+		"deployments":   &appsv1.Deployment{},
+		"statefulsets":  &appsv1.StatefulSet{},
+		"daemonsets":    &appsv1.DaemonSet{},
+		"prometheuses":  &monitoringv1.Prometheus{},
```

Comment:
```suggestion
		"prometheis":  &monitoringv1.Prometheus{},
```
Ref: https://prometheus.io/docs/introduction/faq/#what-is-the-plural-of-prometheus

Reply:
The plural in the CRD definition is `prometheuses` and this is what we have to use here